src/Cake.Common/Tools/GitLink/GitLinkRunner.cs

```diff
@@ -0,0 +1,150 @@
+    public sealed class GitLinkRunner : Tool<GitLinkSettings>
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly ICakeEnvironment _environment;
+        private readonly ICakeLog _log;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GitLinkRunner"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="globber">The globber.</param>
+        /// <param name="log">The logger.</param>
+        public GitLinkRunner(IFileSystem fileSystem, ICakeEnvironment environment, IProcessRunner processRunner, IGlobber globber, ICakeLog log)
+            : base(fileSystem, environment, processRunner, globber)
+        {
+            _fileSystem = fileSystem;
+            _environment = environment;
+            _log = log;
+        }
+
+        /// <summary>
+        /// Update pdb files to link all sources.
+        /// </summary>
+        /// <param name="solutionPath">The directory containing the solution with the pdb files.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(DirectoryPath solutionPath, GitLinkSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            if (solutionPath == null)
+            {
+                throw new ArgumentNullException("solutionPath");
+            }
+
+            Run(settings, GetArguments(solutionPath, settings));
+        }
+
+        private ProcessArgumentBuilder GetArguments(DirectoryPath solutionPath, GitLinkSettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            builder.AppendQuoted(solutionPath.MakeAbsolute(_environment).FullPath);
+
+            if (settings.RepositoryUrl != null)
+            {
+                builder.Append("-u");
+                builder.AppendQuoted(settings.RepositoryUrl.ToString());
+            }
+
+            if (!string.IsNullOrWhiteSpace(settings.SolutionFileName))
+            {
+                builder.Append("-f");
+                builder.AppendQuoted(settings.SolutionFileName);
+            }
+
+            if (!string.IsNullOrWhiteSpace(settings.Configuration))
+            {
+                builder.Append("-c");
+                builder.AppendQuoted(settings.Configuration);
+            }
+
+            if (!string.IsNullOrWhiteSpace(settings.Platform))
+            {
+                builder.Append("-p");
+                builder.AppendQuoted(settings.Platform);
+            }
+
+            if (!string.IsNullOrWhiteSpace(settings.Branch))
+            {
+                builder.Append("-b");
+                builder.AppendQuoted(settings.Branch);
+            }
+
+            if (settings.LogFilePath != null)
```

Comment:
@GeertvanHorrik this is an object of Type FilePath, not just a string.  So either it has something, or it doesn't.